src/components/calendar/drag-drop-provider.tsx

```
@@ -0,0 +1,353 @@
+  const { isDragging, draggedTask, draggedTaskRef, startDrag, cancelDrag } =
+    useDragState();
+  const {
+    listRef,
+    currentScrollOffsetRef,
+    viewportHeightRef,
+    registerListRef,
+    registerViewportHeight,
+    updateCurrentOffset,
+  } = useListRefs();
+  const { undoRef, performUndo } = useUndoState();
+  const { scrollBy, onDragUpdate } = useScrolling(
```

ESLint

[error] 'scrollBy' is assigned a value but never used. Allowed unused vars must match /^_/u.

(unused-imports/no-unused-vars)